src/Publisher.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace Drupal\applenews;
+
+use ChapterThree\AppleNewsAPI\PublisherAPI;
+use Drupal\applenews\Exception\ApplenewsInvalidResponseException;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+
+/**
+ * Applenews publisher manager.
+ */
+class Publisher implements PublisherInterface {
+  use StringTranslationTrait;
+
+  /**
+   * The applenews settings config object.
+   *
+   * @var \Drupal\Core\Config\Config
+   */
+  protected $config;
+
+  /**
+   * Construct the PublisherManager object.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The config factory.
+   */
+  public function __construct(ConfigFactoryInterface $config_factory) {
+    $this->config = $config_factory->get('applenews.settings');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getChannel($channel_id) {
+    $response = $this->publisher()->get('/channels/{channel_id}', ['channel_id' => $channel_id]);
+    return $this->handleResponse($response);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getArticle() {
+
```

Comment:
Any reason why this and the following are empty? Perhaps adding a `@TODO` might help?

Reply:
@juampynr my bad. We are going to fill all these methods before the merge.